modules/factoidaddress/factoidaddress.go

```diff
@@ -0,0 +1,51 @@
+package factoidaddress
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+)
+
+var FactoidAddressPrefix = [2]byte{0x5f, 0xb1}
+
+// Valid returns if the address is a valid factoid address
+func Valid(addr string) error {
+	data := Base58Decode(addr)
+	if len(data) == 0 {
+		return fmt.Errorf("address must be in base58")
+	}
+
+	// prefix + rcd + checksum
+	if len(data) != 2+32+4 {
+		return fmt.Errorf("address is of wrong length")
+	}
+
+	prefix := data[:2]
+	if bytes.Compare(prefix, FactoidAddressPrefix[:]) != 0 {
+		return fmt.Errorf("address has wrong prefix")
+	}
+
+	checksum := data[len(data)-4:]
```

Comment:
since we're making this a module, we should factor out checksum() into its own function

Reply:
:+1: